src/app/search/search-filters/range-numeric.tsx

```diff
@@ -87,6 +87,21 @@ const simpleRangeFilters: FilterDefinition[] = [
       };
     },
   },
+  {
+    keywords: 'weaponlevel',
+    description: tl('Filter.WeaponLevel'),
+    format: 'range',
+    destinyVersion: 2,
+    filter:
+      ({ filterValue }) =>
+      (item) => {
+        if (!item.craftedInfo) {
+          return false;
+        }
+        const compareTo = rangeStringToComparator(filterValue);
+        return compareTo(item.craftedInfo.level || 0);
+      },
+  },
```

Comment:
This is what I meant by hoisting `rangeStringToComparator` up a level:
```ts
filter:
  ({ filterValue }) => {
    const compareTo = rangeStringToComparator(filterValue);
    return (item) => {
      if (!item.craftedInfo) {
        return false;
      }
      return compareTo(item.craftedInfo.level || 0);
    }
  }
```

Reply:
thank you for finding that. a really weird diff section revert in vscode moved this line, but when i got a similar crash i wasn't sure how to repro